File: src/DashTab.Domain/Entities/Floor.cs
using DashTab.Domain.Common;

namespace DashTab.Domain.Entities;

public class Floor : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string BackgroundColor { get; set; } = "#FFFFFF";
    public string? BackgroundImageUrl { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public Tenant Tenant { get; set; } = null!;
    public ICollection<Table> Tables { get; set; } = new List<Table>();
}

File: dashtab_pos/lib/domain/entities/floor.dart
import 'table.dart';

class Floor {
  final String id;
  final String name;
  final String backgroundColor;
  final String? backgroundImageUrl;
  final List<RestaurantTable> tables;

  Floor({
    required this.id,
    required this.name,
    required this.backgroundColor,
    this.backgroundImageUrl,
    required this.tables,
  });

  factory Floor.fromJson(Map<String, dynamic> json) {
    var tablesList = json['tables'] as List? ?? [];
    return Floor(
      id: json['id'],
      name: json['name'],
      backgroundColor: json['backgroundColor'] ?? '#FFFFFF',
      backgroundImageUrl: json['backgroundImageUrl'],
      tables: tablesList.map((t) => RestaurantTable.fromJson(t)).toList(),
    );
  }
}
